swap/protocol_test.go

```diff
@@ -379,13 +392,44 @@ func TestTriggerPaymentThreshold(t *testing.T) {
 		t.Fatalf("Expected cheque contract to be %x, but is %x", debitorSwap.contract.ContractParams().ContractAddress, pending.Contract)
 	}
 
-	debitorSwap.handleConfirmChequeMsg(ctx, creditor, &ConfirmChequeMsg{
-		Cheque: creditor.getPendingCheque(),
+	// we expect a EmitChequeMsg to be sent, then we trigger a ConfirmChequeMsg for the same cheque
+	err = protocolTester.TestExchanges(p2ptest.Exchange{
+		Expects: []p2ptest.Expect{
+			{
+				Code: 1,
+				Msg: &EmitChequeMsg{
+					Cheque: creditor.getPendingCheque(),
+				},
+				Peer: protocolTester.Nodes[0].ID(),
+			},
+		},
+	}, p2ptest.Exchange{
+		Triggers: []p2ptest.Trigger{
+			{
+				Code: 2,
+				Msg: &ConfirmChequeMsg{
+					Cheque: creditor.getPendingCheque(),
+				},
+				Peer: protocolTester.Nodes[0].ID(),
+			},
+		},
 	})
+	if err != nil {
+		t.Fatal(err)
+	}
 
-	// we should now have a cheque
-	if creditor.getLastSentCheque() == nil {
-		t.Fatal("Expected one cheque, but there is none")
+	// we wait until the confirm message has been processed
+loop:
+	for {
+		select {
+		case <-ctx.Done():
+			t.Fatal("Expected one cheque, but there is none")
+		default:
+			if creditor.getLastSentCheque() != nil {
```

Comment:
This actually means that if the remote peer does not confirm the cheque, the node remains "stuck". So if a new cheque would be due and for some reason the confirmation msg never arrived, can the node process it based on the new cumulative amount or something? Or will it reject it

Reply:
As long as there is no confirmation the node will try to resend the current pending cheque once the threshold is reached again (the recipient will send the confirm message even if it has already processed it in the past) and then afterwards send a new cheque for the amount of debt incurred in the meantime. One of the reasons why we don't want to send a new cheque with a higher amount before that is that if the old cheque never was processed correctly by the recipient, then the recipient will consider the difference in `cumulativePayout` compared to the last cheque it has seen as the payment for the honey amount specified in the latest cheque (which doesn't match the expected price and will therefore be rejected). If there is no honey-to-money anymore in the new swap pricing, maybe we can also allow trying to resend with a higher payout.